model_compression_toolkit/target_platform_capabilities/schema/v1.py

```diff
@@ -203,64 +219,92 @@ def clone_and_edit(self, attr_to_edit: Dict[str, Dict[str, Any]] = {}, **kwargs)
         Returns:
             OpQuantizationConfig: Edited quantization configuration.
         """
-
         # Clone and update top-level attributes
-        updated_config = replace(self, **kwargs)
+        updated_config = self.model_copy(update=kwargs)
 
         # Clone and update nested immutable dataclasses in `attr_weights_configs_mapping`
         updated_attr_mapping = {
             attr_name: (attr_cfg.clone_and_edit(**attr_to_edit[attr_name])
-                        if attr_name in attr_to_edit else attr_cfg)
+            if attr_name in attr_to_edit else attr_cfg)
             for attr_name, attr_cfg in updated_config.attr_weights_configs_mapping.items()
         }
 
         # Return a new instance with the updated attribute mapping
-        return replace(updated_config, attr_weights_configs_mapping=updated_attr_mapping)
+        return updated_config.model_copy(update={'attr_weights_configs_mapping': updated_attr_mapping})
 
 
-@dataclass(frozen=True)
-class QuantizationConfigOptions:
+class QuantizationConfigOptions(BaseModel):
     """
     QuantizationConfigOptions wraps a set of quantization configurations to consider during the quantization of an operator.
 
     Attributes:
         quantization_configurations (Tuple[OpQuantizationConfig]): Tuple of possible OpQuantizationConfig to gather.
         base_config (Optional[OpQuantizationConfig]): Fallback OpQuantizationConfig to use when optimizing the model in a non-mixed-precision manner.
     """
-    quantization_configurations: Tuple[OpQuantizationConfig]
+    quantization_configurations: Tuple[OpQuantizationConfig, ...]
     base_config: Optional[OpQuantizationConfig] = None
 
-    def __post_init__(self):
+    # Pydantic v2 configuration for immutability
+    model_config = ConfigDict(frozen=True)
+
+    @model_validator(mode='before')
+    def validate_and_set_base_config(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         """
-        Post-initialization processing for input validation.
+        Validate and set the base_config based on quantization_configurations.
 
-        Raises:
-            Logger critical if quantization_configurations is not a tuple, contains invalid elements, or if base_config is not set correctly.
+        Args:
+            values (Dict[str, Any]): Input data.
+
+        Returns:
+            Dict[str, Any]: Modified input data with base_config set appropriately.
         """
-        # Validate `quantization_configurations`
-        if not isinstance(self.quantization_configurations, tuple):
+        quantization_configurations = values.get('quantization_configurations', ())
```

Comment:
Why return an empty tuple? isn't this an error? I'm unable to follow what is the scenario in which values doesn't include quantization_configurations